packages/valory/connections/abci/connection.py

```diff
@@ -1353,12 +1364,16 @@ def _process_tendermint_params(self) -> None:
         self.use_tendermint = cast(
             bool, self.configuration.config.get("use_tendermint")
         )
-        self.use_grpc = cast(bool, self.configuration.config.get("use_grpc"))
+        if not isinstance(self.use_tendermint, bool):
+            raise TypeError(f"use_tendermint must be bool, got {self.use_tendermint}")
+        self.use_grpc = cast(bool, self.configuration.config.get("use_grpc", False))
 
         if not self.use_tendermint:
             return
```

Comment:
use_grpc was assumed to be a boolean. This was not necessarily so. `(cast` should be used carefully. Since we assume this I decided to add a check, forcing assignment and not silently passing when `None`.